src/bessel.jl

```diff
@@ -590,6 +590,30 @@ function bessely(n::Integer, x::BigFloat)
     return z
 end
 
+"""
+    sphericalbesselj(nu, x)
+
+Spherical bessel function of the first kind at order `nu`, ``j_ν(x)``. This is the non-singular
+solution to the radial part of the Helmholz equation in spherical coordinates.
+"""
+function sphericalbesselj(nu, x::T) where {T}
+    besselj_nuhalf_x = besselj(nu + 1//2, x)
+    if abs(x) ≤ sqrt(eps(real(zero(besselj_nuhalf_x))))
```

Comment:
This is a very suspicious special case. One would expect to get relative accuracy near the origin.

Reply:
I recall testing this and without the special case, one get's NaN near the origin

Reply:
I'm not seeing that:
```julia
julia> x = eps()
2.220446049250313e-16

julia> besselj_nuhalf_x = besselj(1/2,x)
1.188940645460586e-8

julia> √((float(T))(π)/2x) * besselj_nuhalf_x
1.0000000000000018

julia> besselj_nuhalf_x = besselj(3/2,x)
8.79992853002023e-25

julia> √((float(T))(π)/2x) * besselj_nuhalf_x
7.401486830834373e-17
```

Reply:
x has to about 1e-100 range or smaller (if I recall, I can't test it right now)

Reply:
Ahhh, I didn't realise `eps(::Number)` was different than `eps(::Type)`:
```julia
julia> sqrt(eps(Float64))
1.4901161193847656e-8

julia> sqrt(eps(0.0))
2.2227587494850775e-162
```

Reply:
right...although I don't exactly know if this cutoff is good enough or not